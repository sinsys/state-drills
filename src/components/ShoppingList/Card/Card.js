import React from 'react';
import './Card.css';

class Card extends React.Component {

	constructor(props){
		super(props);
		this.state = {
			content: props.content,
			purchased: props.purchased,
			purchasedText: ["Show Again", "Purchase"]
		};
	}

	togglePurchased() {

		this.setState(
			{purchased: !this.state.purchased
		})
		if(this.state.purchased){
			console.log(true);
		}
		console.log(this.state);
	}


	render() {
		const purchase = false;
		const textVal = this.state.purchased ? this.state.purchasedText[0] : this.state.purchasedText[1]
		return (
			<div 
				className={`Card ${this.state.purchased ? " purchased" : "" }`}
			>

				<h3>{this.props.title}</h3>
				<p>{this.props.content}</p>
				<div className="Card-buttons">
					<button
						type="button"
						onClick={() => this.togglePurchased()}
					>
						{textVal}
					</button>
					<button 
						type="button"
						onClick={() => this.props.onClickDelete(this.props.id)}
					>
						Delete
					</button>
					{/* <button
					// 	type="button"
					// 	onClick={() => this.props.onClickSavedCard(this.state.id)}
					// >
					// 	Save for Later
					 </button> */}
				</div>
			</div>
		)		
	}
}

Card.propTypes = {
  onClickDelete: () => {},
  onClickSavedCard: () => {}
}

export default Card;