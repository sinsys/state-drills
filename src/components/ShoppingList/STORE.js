export default {
  lists: [
    {
      id: '1',
      header: 'Daily',
      cardIds: [ 'a', 'b', 'e', 'f', 'g', 'j', 'l', 'm' ],
    },
    {
      id: '2',
      header: 'Weekly',
      cardIds: ['b', 'c', 'd', 'f', 'h', 'i', 'k'],
    },
    {
      id: '3',
      header: 'Monthly',
      cardIds: [ 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm' ],
    },
    {
      id: '4',
      header: 'Seasonal',
      cardIds: [ 'l', 'm' ],
    },
  ],
  allCards: {
    'a': { id: 'a', title: 'First card', content: 'lorem ipsum' },
    'b': { id: 'b', title: 'Second card', content: 'lorem ipsum' },
    'c': { id: 'c', title: 'Third card', content: 'lorem ipsum' },
    'd': { id: 'd', title: 'Fourth card', content: 'lorem ipsum' },
    'e': { id: 'e', title: 'Fifth card', content: 'lorem ipsum' },
    'f': { id: 'f', title: 'Sixth card', content: 'lorem ipsum' },
    'g': { id: 'g', title: 'Seventh card', content: 'lorem ipsum' },
    'h': { id: 'h', title: 'Eighth card', content: 'lorem ipsum' },
    'i': { id: 'i', title: 'Ninth card', content: 'lorem ipsum' },
    'j': { id: 'j', title: 'Tenth card', content: 'lorem ipsum' },
    'k': { id: 'k', title: 'Eleventh card', content: 'lorem ipsum' },
    'l': { id: 'l', title: 'Twelfth card', content: 'lorem ipsum' },
    'm': { id: 'm', title: 'Thirteenth card', content: 'lorem ipsum' },
  }
}