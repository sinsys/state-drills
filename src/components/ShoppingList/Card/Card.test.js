import React from 'react';
import ReactDOM from 'react-dom';
import Card from './Card';
import renderer from 'react-test-renderer';
import STORE from '../STORE.js';

describe('Card component', () => {
	const store = {STORE};
	const props = store.STORE.allCards['a'];

	it('Renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(
			<Card
				id={props.id}
				title={props.title}
				content={props.content}
    	/>, div);
		ReactDOM.unmountComponentAtNode(div);
	});

	it('renders the UI as expected', () => {
		const tree = renderer
			.create(<Card
				id={props.id}
				title={props.title}
				content={props.content}
    	/>)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

})