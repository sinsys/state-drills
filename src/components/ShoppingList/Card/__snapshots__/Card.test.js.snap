// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders the UI as expected 1`] = `
<div
  className="Card "
>
  <h3>
    First card
  </h3>
  <p>
    lorem ipsum
  </p>
  <div
    className="Card-buttons"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Purchased
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      Remove
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      Save for Later
    </button>
  </div>
</div>
`;
